chap1-introduction.tex

```diff
@@ -23,84 +23,69 @@ \chapter{Introduction}\label{ch:intro}
-% ``While we tackled this project mainly to explore the capabilities of a modern formal proof system—at first, to benchmark speed—'' - gonthier2008formal
-% ``We present a new implementation of a reflexive tactic whichsolves equalities in a ring structure inside the Coq system.The efficiencyis improved to a point that we can now prove equalities that were previ-ously beyond reach.'' - Proving2005Benjamin
+However, in all of these tools, there is enormous proof overhead over the lines of code being verified. (Insert graph) 
 
-\todonz{Seems fine ... but is it really unique to proofs?  C++ compilers in the 90s had the same problems}
+To decrease the marginal overhead of proof generation, we may automate the proof generation itself. 
+Drawing experience from case studies in category theory~\cite{category-coq-experience}, parsers~\cite{jgross-masters-thesis}, and generation of low-level cryptographic code~\cite{FiatCryptoSP19}, we present the difficulties in automation of proof generation. 
+In the quest to build automated proof generation tools, we encountered several performance bottlenecks in proof-checking. In this thesis, we present design principles in \autoref{part:design} to avoid these performance bottlenecks, and insights about the proof-checking system that the performance bottlenecks reveal. We present a research prototype of a tool and methodology for achieving acceptable performance at scale in \autoref{part:rewriting}. 
 
 Unlike many other performance domains, in our experience, the time it takes to check proofs as we scale the size of the input is almost always super-linear---quadratic at best, commonly cubic or exponential, occasionally even worse.
 Empirically, this might look like a proof script that checks in tens of seconds on the smallest of toy examples; takes about a minute on the smallest real-world example, which might be twice the size of the toy example; takes twenty hours to generate and check the proof of an example only twice that size; and, on a (perhaps not-quite-realistic) example twice the size of the twenty-hour example, the script might not finish within a year, or even a thousand years---see \autoref{sec:fiat-crypto-codegen-numbers} for more details.
 In just three doublings of input size, we might go from tens of seconds to thousands of years.
 Moreover, in proof assistants, this is not an isolated experience.
 This sort of performance behavior is \emph{typical}.
```

Comment:
\todonz{Seems fine ... but is it really unique to proofs?  C++ compilers in the 90s had the same problems}